src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -188,23 +178,149 @@ function discoverController(
+      }
+    }
+  };
+
+  const stopSyncingAppFilters = syncAppFilters(filterManager, {
+    set: filters => setAppState({ filters: filters }),
+    get: () => getAppFilters(),
+    state$: appStateContainer.state$.pipe(map(state => state.filters)),
+  });
+  // syncs `_g` portion of url with query services
+  const { stop: stopSyncingGlobalStateWithUrl } = syncQuery(data.query, kbnUrlStateStorage);
+
+  $scope.filters = filterManager.getFilters();
+  $scope.onFiltersUpdated = filters => filterManager.setFilters(filters);
+
+  const appStateUnsubscribe = appStateContainer.subscribe(async newState => {
+    const { state: newStatePartial } = splitState(newState);
+    const { state: oldStatePartial } = splitState($scope.state);
+
+    if (!_.isEqual(newStatePartial, oldStatePartial)) {
+      $scope.$evalAsync(async () => {
+        $scope.state = { ...newState };
+
+        // detect changes that should trigger fetching of new data
+        const queryChanged = !_.isEqual(newStatePartial.query, getPreviousAppState().query);
+
+        const currentSort = getSortArray($scope.searchSource.getField('sort'), $scope.indexPattern);
+        const sortChanged = !_.isEqual(currentSort, newStatePartial.sort);
+        const savedQueryChanged = oldStatePartial.savedQuery !== newStatePartial.savedQuery;
+
+        if (savedQueryChanged) {
+          if (
+            newStatePartial.savedQuery &&
+            (!$scope.savedQuery || $scope.savedQuery.id !== newStatePartial.savedQuery)
+          ) {
+            // savedQuery was only changed in url which means it has to be fetched
+            $scope.savedQuery = await data.query.savedQueries.getSavedQuery(
+              newStatePartial.savedQuery
+            );
+            updateStateFromSavedQuery($scope.savedQuery);
+          } else if ($scope.savedQuery && typeof newStatePartial.savedQuery === 'undefined') {
+            $scope.savedQuery = undefined;
+          }
+        }
+        if (queryChanged || sortChanged || savedQueryChanged) {
+          $fetchObservable.next();
+        } else {
+          $scope.$digest();
+        }
+      });
+    }
+  });
+
+  $scope.$watch('state.index', (index, prevIndex) => {
+    if (prevIndex == null || prevIndex === index) return;
+    setAppState({ index });
+    $route.reload();
+  });
+
+  $scope.onRefreshChange = ({ isPaused, refreshInterval }) => {
+    const newInterval = {
+      pause: isPaused,
+      value: refreshInterval,
+    };
+    timefilter.setRefreshInterval(newInterval);
+    $scope.refreshInterval = newInterval;
+  };
+
+  $scope.$watchCollection('state.columns', val => {
+    if (!_.isEqual(val, appStateContainer.get().columns)) {
+      setAppState({ columns: val });
+    }
+  });
+  // update data source when filters update
+  subscriptions.add(
+    subscribeWithScope($scope, filterManager.getUpdates$(), {
+      next: () => {
+        $scope.filters = filterManager.getFilters();
```

Comment:
Seems the `$scope.filters` isn't used, could be removed in both places (line 248)

Reply:
Can't do that, is needed here: https://github.com/elastic/kibana/blob/06d26f6b68aaf9d5958db49d52f75f38044b525c/src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.html#L16